tests/unit/s2n_client_hello_test.c

```diff
@@ -284,6 +284,23 @@ int main(int argc, char **argv)
                 EXPECT_SUCCESS(s2n_config_free(config));
             };
 
+            /* Generate a session id if the negotiated protocol is less than TLS1.3 */
+            {
+                struct s2n_connection *conn;
+                EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+                struct s2n_stuffer *hello_stuffer = &conn->handshake.io;
+                conn->actual_protocol_version = S2N_TLS12;
+
```

Comment:
Should you either also set client_protocol_version or verify that it's >=S2N_TLS13, to make sure you're covering this bug?